src/Stratis.Bitcoin/Utilities/MemoryCache.cs

```diff
@@ -52,8 +52,8 @@ public MemoryCache(int maxItemsCount, IEqualityComparer<TKey> comparer = null)
 
             this.maxItemsCount = maxItemsCount;
 
-            this.cache = new Dictionary<TKey, LinkedListNode<CacheItem<TKey, TValue>>>(this.maxItemsCount, comparer);
-            this.keys = new LinkedList<CacheItem<TKey, TValue>>();
+            this.cache = new Dictionary<TKey, LinkedListNode<CacheItem>>(this.maxItemsCount, comparer);
+            this.keys = new LinkedList<CacheItem>();
```

Comment:
Don't use the name `mutex` its confusing, just a `lockObject`  is enough.

Reply:
Fixed